command/agent/job_endpoint.go

```diff
@@ -699,25 +699,33 @@ func (s *HTTPServer) JobsParseRequest(resp http.ResponseWriter, req *http.Reques
 	return jobStruct, nil
 }
 
-func regionForJob(job *api.Job, queryRegion, agentRegion string) (string, string) {
+func regionForJob(job *api.Job, queryRegion, writeRegion, agentRegion string) (string, string) {
 	var requestRegion string
 	var jobRegion string
 
-	if queryRegion != "" {
-		requestRegion = queryRegion
-		jobRegion = queryRegion
+	// Region in http request body takes precedence
```

Comment:
i'm not sure if this is correct... the way that the code was previously performing this at the very end suggests to me that (whether intentional or not), the http url parameter took precedence: 

https://github.com/hashicorp/nomad/pull/8347/commits/99529f233f827041d27b96a082b1a2b439c1c997#diff-873734b35e73e20c8685472523e5a532R424

Reply:
I believe that's part of the original bug... the docs currently say:

> If the job has specified the region, the -region flag and $NOMAD_REGION environment variable are overridden and the job's region is used.

Reply:
Per a sidebar discussion with @cgbaker I'm going to restore the existing behavior here for 0.12.0 and open a new issue to update the documentation and/or behavior in an upcoming release.

Reply:
That new issue: https://github.com/hashicorp/nomad/issues/8365